code/core/src/manager/components/sidebar/Search.tsx

```diff
@@ -272,9 +278,26 @@ export const Search = React.memo<{
     [api]
   );
 
-  const onInputValueChange = useCallback((inputValue: string, stateAndHelpers: any) => {
-    showAllComponents(false);
-  }, []);
+  const onInputValueChange = useCallback(
+    (inputValue: string, stateAndHelpers: ControllerStateAndHelpers<DownshiftItem>) => {
+      showAllComponents(false);
+      const isBrowsing = !stateAndHelpers.isOpen && document.activeElement !== inputRef.current;
+      api.setQueryParams({
+        [FILTER_KEY]: isBrowsing ? undefined : inputValue,
+      });
+      const params = new URLSearchParams(window.location.search);
+      if (window.history.replaceState) {
+        if (inputValue) {
+          params.set(FILTER_KEY, inputValue);
+        } else {
+          params.delete(FILTER_KEY);
+        }
+        const paramsString = params.size > 0 ? `?${params.toString()}` : '';
+        window.history.replaceState({}, '', paramsString);
+      }
```

Comment:
this isn't needed (maybe it was once), I'm pretty certain `api.setQueryParams()` already does this.

```suggestion
```

Reply:
im not sure it actually does unless im just looking in the wrong place. i expected it to for sure but this doesn't seem that way:

https://github.com/storybookjs/storybook/blob/6388f93b1f3517b8b0dad3dbd05aa5d5313ee5ae/code/core/src/manager-api/modules/url.ts#L192-L208

Reply:
You're looking at the right place, but the code flow is a long and winding road 😅

1. `setQueryParams` emits the `UPDATE_QUERY_PARAMS` event on line 206
https://github.com/storybookjs/storybook/blob/6388f93b1f3517b8b0dad3dbd05aa5d5313ee5ae/code/core/src/manager-api/modules/url.ts#L192-L208

2. The `PreviewWithSelection` instance listens for that event 
https://github.com/storybookjs/storybook/blob/6388f93b1f3517b8b0dad3dbd05aa5d5313ee5ae/code/core/src/preview-api/modules/preview-web/PreviewWithSelection.tsx#L107

3. ... and calls `setQueryParams` on the `SelectionStore` instance. https://github.com/storybookjs/storybook/blob/4419ab2fce100d287ec8b5aa01db9c81151f0e0d/code/core/src/preview-api/modules/preview-web/PreviewWithSelection.tsx#L235-L237

4. The `selectionStore` happens to be a `UrlStore` that does `history.replace` in the end. https://github.com/storybookjs/storybook/blob/4419ab2fce100d287ec8b5aa01db9c81151f0e0d/code/core/src/preview-api/modules/preview-web/UrlStore.ts#L106-L110

Reply:
Aha thanks a lot for the explanation, makes sense to me!

Reply:
You were right I was wrong!
The `PreviewWithSelection` that listens for the event is on the preview-side (the iframe that shows the story), so `setQueryParams` would only set the query params in the iframe, not in the actual top-level manager.

I changed it to use `applyQueryParams`, which fixes it.